components/cu_transform/src/transform.rs

```diff
@@ -0,0 +1,371 @@
+use cu29::clock::{CuTime, CuTimeRange};
+use cu_spatial_payloads::Transform3D;
+use dashmap::DashMap;
+use serde::{Deserialize, Serialize};
+use std::collections::VecDeque;
+use std::fmt::Debug;
+use std::sync::{Arc, RwLock};
+
+const DEFAULT_CACHE_SIZE: usize = 100;
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct StampedTransform<T: Copy + Debug + 'static> {
+    pub transform: Transform3D<T>,
+    pub stamp: CuTime,
+    pub parent_frame: String,
+    pub child_frame: String,
+}
+
+/// Internal transform buffer that holds ordered transforms between two frames
+#[derive(Clone, Debug, Serialize, Deserialize)]
+struct TransformBufferInternal<T: Copy + Debug + 'static> {
+    transforms: VecDeque<StampedTransform<T>>,
+    max_capacity: usize,
+}
+
+/// Thread-safe wrapper around a transform buffer with concurrent access
+#[derive(Clone)]
+pub struct TransformBuffer<T: Copy + Debug + 'static> {
```

Comment:
can we make that constant size too? because all those objects are really small, few bytes, we could have them in the stack, in the state of a task etc...